plugin/src/android/src/main/java/com/mparticle/cordova/MParticleCordovaPlugin.java

```diff
@@ -698,5 +760,35 @@ private static String ConvertPromotionActionType(int promotionActionType) {
                 return Promotion.CLICK;
         }
     }
+
+    private static GDPRConsent ConvertGDPRConsent(JSONObject map) throws JSONException {
+        String dateStr = obj.getString("timestamp");
```

Comment:
Same here, using `obj.optString("timestamp")` then null-checking will allow us to gracefully handle malformed input without crashing..right now this will generate a JSONException if the user leaves out the "timestamp" value